dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -0,0 +1,698 @@
+        ]
+        fq_loads_table_name = self.make_qualified_table_name(self.schema.loads_table_name)
+        write_disposition = self.schema.get_table(self.schema.loads_table_name).get(
+            "write_disposition"
+        )
+        upload_batch(
+            records,
+            db_client=self.db_client,
+            table_name=fq_loads_table_name,
+            write_disposition=write_disposition,
+        )
+
+    def restore_file_load(self, file_path: str) -> LoadJob:
+        return EmptyLoadJob.from_file_path(file_path, "completed")
+
+    def start_file_load(self, table: TTableSchema, file_path: str, load_id: str) -> LoadJob:
+        return LoadLanceDBJob(
+            self.schema,
+            table,
+            file_path,
+            type_mapper=self.type_mapper,
+            db_client=self.db_client,
+            client_config=self.config,
+            model_func=self.model_func,
+            fq_table_name=self.make_qualified_table_name(table["name"]),
+        )
+
+    def table_exists(self, table_name: str) -> bool:
+        return table_name in self.db_client.table_names()
+
+    def _from_db_type(
+        self, wt_t: pa.DataType, precision: Optional[int], scale: Optional[int]
+    ) -> TColumnType:
+        return self.type_mapper.from_db_type(cast(pa.DataType, wt_t), precision, scale)
+
+
+class LoadLanceDBJob(LoadJob):
+    arrow_schema: TArrowSchema
+
+    def __init__(
+        self,
+        schema: Schema,
+        table_schema: TTableSchema,
+        local_path: str,
+        type_mapper: LanceDBTypeMapper,
+        db_client: DBConnection,
+        client_config: LanceDBClientConfiguration,
+        model_func: TextEmbeddingFunction,
+        fq_table_name: str,
+    ) -> None:
+        file_name = FileStorage.get_file_name_from_file_path(local_path)
+        super().__init__(file_name)
+        self.schema: Schema = schema
+        self.table_schema: TTableSchema = table_schema
+        self.db_client: DBConnection = db_client
+        self.type_mapper: TypeMapper = type_mapper
+        self.table_name: str = table_schema["name"]
+        self.fq_table_name: str = fq_table_name
+        self.unique_identifiers: Sequence[str] = list_unique_identifiers(table_schema)
+        self.embedding_fields: List[str] = get_columns_names_with_prop(table_schema, VECTORIZE_HINT)
+        self.embedding_model_func: TextEmbeddingFunction = model_func
+        self.embedding_model_dimensions: int = client_config.embedding_model_dimensions
+        self.id_field_name: str = client_config.id_field_name
+        self.write_disposition: TWriteDisposition = cast(
+            TWriteDisposition, self.table_schema.get("write_disposition", "append")
+        )
+
+        records: List[DictStrAny]
+        with FileStorage.open_zipsafe_ro(local_path) as f:
+            try:
+                records = json.load(f)
+            except JSONDecodeError:
+                # If parsing as a single object fails, try a line at a time.
```

Comment:
you don't need the try except block here, just always go line by line

Reply:
Oh yes its already wrapped in a `JSONDecodeError` 🤦

Thanks! Fixed